include/cuco/detail/open_addressing/functors.cuh

```diff
@@ -44,9 +47,64 @@ struct get_slot {
   {
     auto const window_idx = idx / StorageRef::window_size;
     auto const intra_idx  = idx % StorageRef::window_size;
-    return storage_[window_idx][intra_idx];
+    if constexpr (HasPayload) {
+      auto const& [first, second] = storage_[window_idx][intra_idx];
+      return thrust::make_tuple(first, second);
```

Comment:
Why do we wanna repack the slot's content which is already a `cuco::pair` into a `thrust::tuple`?

Reply:
Good catch. This is only needed by `retrieve_all` where our output type is a zip iterator for map (equivalent to thrust::tuple):
https://github.com/NVIDIA/cuCollections/blob/fabd885327dc1662e1a63d9b8870ebf888893c09/include/cuco/detail/open_addressing/open_addressing_impl.cuh#L566-L573
And to assign a tuple with cub copy_if, the input type must be a tuple as well.

Reply:
> we should avoid using it anywhere in the hot path since repacking might hurt performance.

This functor is only used by `retrieve_all` so we should be good.